plugins/pluginmanagers/lazy.nix

```
@@ -1,222 +1,204 @@
-            path = p.pkg;
-          };
-      processDependencies =
-        if plugin ? dependencies && plugin.dependencies != null then
-          builtins.concatMap processPlugin plugin.dependencies
-        else
-          [ ];
-    in
-    [ (mkEntryFromDrv plugin) ] ++ processDependencies;
-
-  processedPlugins = builtins.concatLists (builtins.map processPlugin lazyPlugins);
-  lazyPath = pkgs.linkFarm "lazy-plugins" processedPlugins;
-in
-{
-  options = {
-    plugins.lazy = {
-      enable = mkEnableOption "lazy.nvim";
+          plugin;
 
-      gitPackage = lib.mkPackageOption pkgs "git" {
-        nullable = true;
-      };
+      lazyPluginCoercableType = with types; either str package;
 
-      plugins =
+      lazyPluginType =
         with types;
-        let
-          pluginType = either package (submodule {
-            options = {
-              dir = helpers.mkNullOrOption str "A directory pointing to a local plugin";
-
-              pkg = mkOption {
-                type = package;
-                description = "Vim plugin to install";
-              };
+        types.coercedTo lazyPluginCoercableType coerceToLazyPluginSpec (
+          submodule (
+            { config, ... }:
+
+            let
+              cfg = config;
+            in
+            {
+              freeformType = attrsOf anything;
+              options = {
+                __unkeyed = mkNullOrOption str ''
+                  The "unkeyed" attribute is the plugin's short plugin url.
+                  It ill be expanded using `config.git.url_format`. It can
+                  also be a url or dir.
+                '';
+
+                dir = mkNullOrOption str "A directory pointing to a local plugin";
+
+                url = mkNullOrOption str "A custom git url where the plugin is hosted";
+
+                pkg = mkNullOrOption package "Vim plugin to install";
+
+                name = mkNullOrOption str "Name of the plugin to install";
+
+                dev = defaultNullOpts.mkBool false ''
+                  When true, a local plugin directory will be used instead.
+                  See config.dev
+                '';
+
+                lazy = defaultNullOpts.mkBool true ''
+                  When true, the plugin will only be loaded when needed.
+                  Lazy-loaded plugins are automatically loaded when their Lua modules are required,
+                  or when one of the lazy-loading handlers triggers
+                '';
+
+                enabled = defaultNullOpts.mkStrLuaFnOr bool "`true`" ''
+                  When false then this plugin will not be included in the spec. (accepts fun():boolean)
+                '';
+
+                cond = defaultNullOpts.mkStrLuaFnOr bool "`true`" ''
+                  When false, or if the function returns false,
+                  then this plugin will not be loaded. Useful to disable some plugins in vscode,
+                  or firenvim for example. (accepts fun(LazyPlugin):boolean)
```

```suggestion
                  When `false`, or if the function returns false,
                  then this plugin will not be loaded. 
                  
                  This can be used to disable some plugins in vscode or firenvim, for example.
```

Wonder if the last part should just be an option example.